### ISSUE TITLE
Reports issue

### ISSUE THREAD

**Issue by arius28** [mautic/mautic#13446]
### Mautic Version

5.0.x series

### Way of installing

I downloaded a release from https://www.mautic.org/mautic-releases

### PHP version

8.1

### What browsers are you seeing the problem on?

Not relevant

### What happened?

Hello,

I was directed from the forum to report my issue here. First I want to say the I initially installed mautic by downloading a release from github and recently updated using the command line to 5.0.3.

Now on to the issue. When I try to add a filter to a report (data source: Emails sent) it crashes while saving. It seems to save it but crashes afterwards. If I re-edit the report the filter is there and if I remove it and save again it saves without error. If I do leave the filter in then when I try to view the report it does not open. When i remove the filter I can view the report.

On the developer tools side I get the following errors when hitting save & close:
![Screenshot 2024-02-22 151004](https://github.com/mautic/mautic/assets/12183504/ee93ca2c-557f-458d-9793-b7df528b4378)

I have attached entries from the log file. I don’t know which of them are related because I also had a problem with sending due to a problem with the email dsn configuration. but the one with 
`Uncaught Error: Too few arguments to function Doctrine\DBAL\Query\Expression\CompositeExpression`
seems to be the related because it appears right after I try to save.

Steps I have tried to fix the problem:
Don’t know what to try. I only tried clearing the cache just in case something was corrupt there.

### How can we reproduce this issue?

Step 1: Create a report with data source emails sent
Step 2: Add for example contact email and campaign in the columns
Step 3: Add a filter. Column: campaign Condition: like or equal
Step 4 Save and close

### Relevant log output

```shell
[2024-02-21T00:40:05.504900+00:00] php.CRITICAL: Fatal Error: Allowed memory size of 134217728 bytes exhausted (tried to allocate 20480 bytes) {“exception”:“[object] (Symfony\Component\ErrorHandler\Error\OutOfMemoryError(code: 0): Error: Allowed memory size of 134217728 bytes exhausted (tried to allocate 20480 bytes) at /var/www/vhosts/klett.gr/comm.klett.gr/app/bundles/LeadBundle/Helper/CustomFieldValueHelper.php:24)”} {“hostname”:“chost”,“pid”:2414846}

[2024-02-21T14:13:25.442920+00:00] php.CRITICAL: Fatal Error: Allowed memory size of 134217728 bytes exhausted (tried to allocate 196608 bytes) {“exception”:“[object] (Symfony\Component\ErrorHandler\Error\OutOfMemoryError(code: 0): Error: Allowed memory size of 134217728 bytes exhausted (tried to allocate 196608 bytes) at /vendor/doctrine/dbal/src/Driver/PDO/Statement.php:130)”} {“hostname”:“chost”,“pid”:2386549}

[2024-02-21T15:26:08.869788+00:00] php.CRITICAL: Uncaught Error: Too few arguments to function Doctrine\DBAL\Query\Expression\CompositeExpression::and(), 0 passed in /app/bundles/ReportBundle/Builder/MauticReportBuilder.php on line 468 and exactly 1 expected {“exception”:“[object] (ArgumentCountError(code: 0): Too few arguments to function Doctrine\DBAL\Query\Expression\CompositeExpression::and(), 0 passed in /app/bundles/ReportBundle/Builder/MauticReportBuilder.php on line 468 and exactly 1 expected at /vendor/doctrine/dbal/src/Query/Expression/CompositeExpression.php:65)”} {“hostname”:“chost”,“pid”:2583706}
```

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

<br /><hr>
Care about this issue? Want to get it resolved sooner? If you are a <a href='https://www.mautic.org/become-a-member-of-mautic'>member of Mautic</a>, you can add some funds to the <a href='https://opencollective.com/mautic/projects/bounties'>Bounties Project</a> so that the person who completes this task can claim those funds once it is merged by a member of the core team! Read the docs <a href='https://contribute.mautic.org/product-team/mautic-bounty-programme'>here.</a>

**Comment by patrykgruszka** [comment]
This issue should be already fixed in #12569

**Comment by RCheesley** [comment]
@arius28 assuming this is fixed as mentioned above I'm closing this issue. Thanks for reporting it and sorry nobody came through to triage it sooner.
